### PR TITLE
test: fix retry logic in the event counter tests

### PR DESCRIPTION
**Pull request by cre8ivejp**
There was a bug in the retry logic, making it retry less than it was supposed to. In that case, it was 30 times.
Also, I made some improvements around the timeout and the retry logic.
Now, after retrying, it won't fail. Instead, it will return the error the caller handles and proceed accordingly.

### CODE REVIEW

File: test/e2e/eventcounter/eventcounter_test.go [hunk 1]
```diff
@@ -537,7 +591,17 @@ func TestExperimentResult(t *testing.T) {
 			t.Fatalf("retry timeout")
 		}
 		time.Sleep(10 * time.Second)
-		resp := getExperimentResult(t, ecClient, experiment.Id)
+
+		resp, err := getExperimentResult(t, ecClient, experiment.Id)
+		if err != nil {
+			st, _ := status.FromError(err)
+			if st.Code() != codes.NotFound {
```
Comment (cre8ivejp) [inline test/e2e/eventcounter/eventcounter_test.go:598]: We will only retry if the experiment is not ready.

File: test/e2e/eventcounter/eventcounter_test.go [hunk 2]
```diff
@@ -2099,67 +2291,84 @@ func getFeature(t *testing.T, client featureclient.Client, featureID string) *fe
...
+	for i := 0; i < numRetries; i++ {
+		response, err = c.GetEvaluationTimeseriesCount(
+			ctx,
+			req,
+		)
+		if err == nil {
+			break
+		}
+		st, _ := status.FromError(err)
+		if st.Code() != codes.Unavailable {
```
Comment (cre8ivejp) [inline test/e2e/eventcounter/eventcounter_test.go:2336]: It only retries for all the requests if the server is unavailable due to pods rescheduling, etc.

File: test/e2e/eventcounter/eventcounter_test.go [hunk 3]
```diff
@@ -53,8 +53,8 @@ import (
 
 const (
 	prefixTestName = "e2e-test"
-	timeout        = 20 * time.Second
-	retryTimes     = 30
+	timeout        = 60 * time.Second
```
Comment (cre8ivejp) [inline test/e2e/eventcounter/eventcounter_test.go:56]: During the tests, I saw a few times a context deadline was exceeded while inserting data in BigQuery. So, I changed it to 60 seconds as the dev environment is configured.

File: test/e2e/eventcounter/eventcounter_test.go [hunk 4]
```diff
@@ -2001,26 +2189,28 @@ func getExperimentResult(t *testing.T, c ecclient.Client, experimentID string) *
...
+	numRetries := 3
 	for i := 0; i < numRetries; i++ {
 		response, err = c.GetExperimentResult(ctx, req)
 		if err == nil {
 			break
 		}
 		st, _ := status.FromError(err)
-		if st.Code() == codes.Internal {
-			t.Fatal(err)
+		if st.Code() != codes.Unavailable {
```
Comment (Ubisoft-potato) [inline test/e2e/eventcounter/eventcounter_test.go:2199]: I think we can also retry if the `st.Code()` is **Internal**. For example, we can return `InternalServerError` in the `GetExperimentResult` [function](https://github.com/bucketeer-io/bucketeer/blob/main/pkg/eventcounter/api/api.go#L761-L764), and this `InternalServerError` could also caused by downstream unavailable.

So this condition can be `st.Code() != codes.Unavailable && st.Code() != codes.Internal`.

`codes.Internal` Doc:
```golang
        // Internal errors. Means some invariants expected by underlying
	// system has been broken. If you see one of these errors,
	// something is very broken.
	//
	// This error code will be generated by the gRPC framework in several
	// internal error conditions.
        Internal Code = 13

```

**Comment by Ubisoft-potato** [comment]
LGTM👍!

**Pull request merged**